### PR TITLE
Project structure

### PR DESCRIPTION
**Pull request by rdiazv**
# Explanation
This tries to separate concerns internally while developing components, always maintaining consistency across all of them.

## Styles
Guidelines and helpers that can be used anywhere, even without components.

### Core
SASS variables, mixins and functions used as guidelines to build every component. None of this files generates input if imported, so is safe to include in multiple places.

### Scaffold
[Resets](http://meyerweb.com/eric/tools/css/reset/) all the browser default styles and then applies some minimum rules based on the core variables.

This properties are set by the scaffold:

- Text color.
- Base font size.
- Base line height.

### Utils
Helper classes for common CSS requirements, ie, changing text alignment, managing overflow, etc. This classes allows a more declarative way to lay down components.

All this util classes must be either non-opinionated or based on core variables to allow customization. This seeks consistency across our component and custom ones built by the user.

Some examples:

```sass
@import '~core/variables'

// GOOD. Non-opinionated. Left aligned text is the same in any project.
.text-align-left {
  text-align: left;
}

// GOOD. Based on core variables. The user can customize the variables values.
.type-title {
  font-size: $font-size-title;
  line-height: $line-height-title;
}

// BAD. What is small?
.margin-small {
  margin: 5px;
}
```

## Components
React components that follows all the styles guidelines. This components are based entirely on the core variables and helper classes.

# Changes
- You can now import the core variables, mixins and functions with `@import '~portrait/lib/scss/core'`.
- You can now import the scaffold styles with `@import '~portrait/lib/scss/scaffold'`.
- You can now import the helper classes with `@import '~portrait/lib/scss/utils'`.
- Added `Grid` component (actually it already existed with a *noop* implementation). This component was added to put in practice all the previously explained changes.

**Note:** All the new style imports are opt-in and aren't included on the css build.

# Development changes
- Examples can be render on a collapsible container using `collapsible: true|false` (`false` by default). The initial state can be specified with `expanded: true|false` (`true` by default).
- Examples can be rendered with or without the component declaration. This allows to write more complex examples if required (split markup, manage state, etc).

Inline examples declares only the `render` code.

```yml
inline: true # Default value. Can be omitted.
code: >
  <div />
```

Full component examples must declare and render the component manually on the `mountNode` node (automatically injected).

```yml
inline: false
code: >
  const Example = () => <div />
 
  ReactDOM.render(<Example />, mountNode)
```

### CODE REVIEW

**Comment by rdiazv** [comment]
Esto es un draft. La idea es discutir y afinar los conceptos que estoy exponiendo. Si a alguien se le ocurre otra organización del proyecto, pronúnciese xd.

File: docs/components/Example.js [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
   componentName: PropTypes.string.isRequired,
   description: PropTypes.string,
+  expanded: PropTypes.bool,
+  inline: PropTypes.bool,
+}
+
+Example.defaultProps = {
+  collapsible: false,
+  expanded: true,
+  inline: true,
```
Comment (cristianferrarig) [inline docs/components/Example.js]: para que es esto?

File: docs/theme.scss [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+
+  .CodeMirror-scroll {
+    overflow: visible !important;
+  }
+
+  .playgroundCode {
+    order: 2;
+  }
+
+  .playgroundPreview {
```
Comment (cristianferrarig) [inline docs/theme.scss]: Estas clases deberían quedar fuera del scope de `.playground` para evitar que compile
`.playground .playgroundCode {...}` la clase puede vivir por si misma `.playgroundCode  {...}`

File: src/styles/scaffold/_reset.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/* http://meyerweb.com/eric/tools/css/reset/
```
Comment (cristianferrarig) [inline src/styles/scaffold/_reset.scss]: Y normalize ya no?

File: docs/components/Example.js (hunk 1 above)
Comment (rdiazv) [inline docs/components/Example.js]: Es el modo en que escribes los ejemplos. Si es `inline` entonces defines solo lo que va dentro del `render`, es para ejemplos simples. Si no es `inline` entonces puedes escribir componentes completos, con estado, etc, y debes renderear manualmente en el dom.

- `inline: true`

```jsx
<div>Some example</div>
```

- `inline: false`

```jsx
const Example = () =>
  <div>Some example</div>

ReactDOM.render(<Example />, mountNode)
```

File: src/styles/scaffold/_reset.scss (hunk 3 above)
Comment (rdiazv) [inline src/styles/scaffold/_reset.scss]: Si, no sé, a mi me da igual en realidad. Investigué cuál de las miles de cuestiones para normalizar los css era mejor y en varios lados recomendaban ese. No puedo dar una opinión objetiva porque no probé personalmente en todos los navegadores, así que con fe no más xD.

File: docs/theme.scss (hunk 2 above)
Comment (rdiazv) [inline docs/theme.scss]: Corregido en 2909a156293d0d695800c1b804d148e621661bbf.

**Comment by cristianferrarig** [comment]
**Tools**
 - Color
 - Maps

**Layout**
- Breakpoints
- Grid & Spacing
- Layers
- Layout Stacked/Flee/Table

**Core**
- Color
- Typography
- Shadows
- Motion
- Borders
- SR only

**Common**
- Normalize
- Scafolding
- Typography
- Links
- Forms
- Images
- Tables

**Components**
- Buttons

**Pull request merged**